src/System.Management.Automation/CoreCLR/CorePsAssemblyLoadContext.cs

```diff
@@ -194,6 +198,50 @@ internal IEnumerable<Assembly> GetAssembly(string namespaceQualifiedTypeName)
             return null;
         }
 
+        /// <summary>
+        /// If a managed dll has native dependencies the handler will try to find these native dlls.
+        ///     1. Gets the managed.dll location (folder)
+        ///     2. Based on OS name and architecture name builds subfolder name where it is expected the native dll resides:
+        ///     3. Loads the native dll
+        ///
+        ///     managed.dll folder
+        ///                     |
+        ///                     |--- 'win-x64' subfolder
+        ///                     |       |--- native.dll
+        ///                     |
+        ///                     |--- 'win-x86' subfolder
+        ///                     |       |--- native.dll
+        ///                     |
+        ///                     |--- 'win-arm' subfolder
+        ///                     |       |--- native.dll
+        ///                     |
+        ///                     |--- 'win-arm64' subfolder
+        ///                     |       |--- native.dll
+        ///                     |
+        ///                     |--- 'linux-x64' subfolder
+        ///                     |       |--- native.dll
+        ///                     |
+        ///                     |--- 'linux-x86' subfolder
+        ///                     |       |--- native.dll
+        ///                     |
+        ///                     |--- 'linux-arm' subfolder
+        ///                     |       |--- native.dll
+        ///                     |
+        ///                     |--- 'linux-arm64' subfolder
+        ///                     |       |--- native.dll
+        ///                     |
+        ///                     |--- 'osx' subfolder
+        ///                     |       |--- native.dll
+        /// </summary>
+        internal static IntPtr NativeDllHandler(Assembly assembly, string libraryName)
+        {
+            System.Diagnostics.Debugger.Break();
+            var folder = Path.GetDirectoryName(assembly.Location);
+            var searchPath = Path.Combine(folder, NativeSubFolderName,libraryName);
+
+            return NativeLibrary.Load(searchPath, assembly, null);
```

Comment:
Should this use `NativeLibrary.Load(string libraryPath)`?

Reply:
I guess current one will be more smart and load native assembly from managed assembly path too that @vexx32 ask.

Reply:
You already calculated the full path to the native library, which is based on the assembly path, isn't it?
So it seems to me `NativeLibrary.Load(string libraryPath)` would be sufficient.

Reply:
I see your point.

Done